vllm/worker/vineyard_llm_cache.py

```diff
@@ -520,7 +514,8 @@ def update_seq_kv_caches(
             ) = update_args
         if update_token_size <= 0:
             # restore the seq_group_metadata's and seq's metadata
-            self._update_seq_group_metadata(seq_group_metadata, -matched[seq_id])
+            if seq_group_metadata is not None
+              self._update_seq_group_metadata(seq_group_metadata, -matched[seq_id])
             return
 
         if self.enable_async_update:
```

Comment:
just realized that when TP is enabled, this part needs go after the broadcast below, could you please make this change in your PR?